symphony-bdk-core/src/main/java/com/symphony/bdk/core/retry/RetryWithRecovery.java

```diff
@@ -97,6 +98,31 @@ protected T executeOnce() throws Throwable {
     }
   }
 
+  /**
+   * This methods check if the type of exception thrown in the retry and depending on that it wraps the Exception
+   * in a {@link RuntimeException} with an error message explaining the cause of the issue.
+   * @param t RuntimeException found
+   * @param address that is not reachable
+   * @return RuntimeException to be thrown
+   */
+  public static String networkIssueMessageError(Throwable t, String address) {
+    String messageError = "An unknown error occurred. Please check below for more information: ";
+    String service = ApiClientFactory.getServiceNameFromBasePath(address).toString();
+    if (t.getCause() instanceof SocketTimeoutException) {
+      messageError = String.format(
+          "Timeout occurred while trying to connect to the \"%s\" at the following address: %s. "
+              + "Please check that the address is correct. Also consider checking your proxy/firewall connections.", service, address);
+      log.error(messageError);
```

Comment:
I put the logger as well because when testing with spring boot project, this exception will be actually visible after a lot of "could not inizialize bean" exceptions. This way instead you can first see this error from the logs and then you have the all stack trace of the exception thrown. Though in this way is was easier to find the cause quickly.

Reply:
I'm removing them, since now we will log the message at each retry the information will already be there